cv_next/app/feed/components/feed.tsx

```diff
@@ -1,82 +1,130 @@
+  const page = useRef<number>(
+    cvsContextConsumer.cvs?.length
+      ? cvsContextConsumer.page
+      : Definitions.PAGINATION_INIT_PAGE_NUMBER
+  )
+  const [loadMore, setLoadMore] = useState(true)
+  const [filters, setFilters] = useState<filterObj>({
+    searchValue: "",
+    categoryId: null
+  })
+
+
+  /**
+   * Trigger pagination when this element comes into view.
+   *
+   * @param {Function} callbackTrigger - A function that triggers pagination when called
+   * @return {JSX.Element} A div element with a ref for triggering pagination
+   */
+  function TriggerPagination({
+    callbackTrigger,
+  }: {
+    callbackTrigger: () => Promise<void>
+  }) {
+    const [triggerRef, inView] = useInView()
+
+    useEffect(() => {
+      if (inView) {
+        callbackTrigger()
+      }
+    }, [callbackTrigger, inView])
+
+    return <div ref={triggerRef}></div>
+  }
 
   useEffect(() => {
-    //update cv-related refs
-    cvsRef.current = cvs;
-    lastCvRef.current = cvs?.[cvs?.length - 1]?.id;
-  }, [ cvs ]);
-  
+    cvsRef.current = cvs
+  }, [cvs])
+
   const fetchCvsCallback = useCallback(async () => {
-    
-    //fetch cvs and handle loading states
     if (loadMore) {
-      const fetchedCvs = await fetchCvs({lastId: lastCvRef.current});
-      
-      if (fetchedCvs && fetchedCvs?.length > 0) {
-        if (fetchedCvs[fetchedCvs.length - 1]?.id === lastCvRef.current) {
-          setLoadMore(false);
-        } else {
-          setCvs((prevCvs) => [ ...(prevCvs?.length ? prevCvs : []), ...fetchedCvs ]);
-        }
+      const nextPage = page.current + 1
+      const response = await fetchCvsForFeed({ page: nextPage, filters: filters })
+      if (response && response.cvs.length > 0) {
+        setCvs((prevCvs) => [...prevCvs, ...response.cvs])
+        page.current = nextPage
       } else {
-        setLoadMore(false);
+        setLoadMore(false)
       }
     }
-  }, [ loadMore ]);
+  }, [loadMore, filters])
 
   useEffect(() => {
     //before unmount, save current cvs state to context for smoother navigation
     //TODO: add time-based revalidation for context
     return () => {
       if (cvsRef.current) {
-        cvsDispatchContextConsumer({ type: `replace`, payload: cvsRef.current });
+        cvsDispatchContextConsumer({
+          type: `replace`,
+          payload: { cvs: cvsRef.current, page: page.current },
+        })
       }
     }
-  }, [ cvsDispatchContextConsumer ]);
+  }, [cvsDispatchContextConsumer])
 
   const forceReload = () => {
-    cvsDispatchContextConsumer({type: `reset`, payload: []});
-    setCvs([]);
-    setLoadMore(true);
-  };
+    cvsDispatchContextConsumer({
+      type: `reset`,
+      payload: {
+        cvs: [],
+        page: Definitions.PAGINATION_INIT_PAGE_NUMBER,
+      },
+    })
+    setCvs([])
+    page.current = Definitions.PAGINATION_INIT_PAGE_NUMBER
+    setLoadMore(true)
+  }
+
+  const onFilterChange = (filters: filterObj) => {
+    setFilters(filters)
+    forceReload()
```

Comment:
Small potential for race with state. Probably needs testing that we currently don't have (do we have filters?), but the `forceReload` function will get called without you knowing if the filters state has already changed.

Reply:
We have category-based filters and I think @nevogithub added multi-category filters

Reply:
there suppose to be a search filter too

Reply:
What I meant but the "do we have filters" is "can we actually test the thought I had" :)
The point is, there's a possible race for the filter state and the force reload functionality. You'd either want to make a synchronous behavior for the filters state change (absolute hell, don't try and it's an anti-pattern for react), or more appropriately replace it with a `useEffect` and add the filters to the deps array.

Reply:
I see
Yes we have filters(both for categories and for a description of a CV and it does seem to work fine)

Reply:
Tagging here just t see if a follow up is needed here @YechiamW3

Reply:
I think that we can leave it, but there's still a racing risk here.
The best way to solve it is to do as I've suggested.

Reply:
Added your suggestion